src/main/scala/io/iohk/ethereum/ledger/BlockExecution.scala

```diff
@@ -57,10 +59,19 @@ class BlockExecution(
   /** Executes a block (executes transactions and pays rewards) */
   private def executeBlock(block: Block): Either[BlockExecutionError, BlockResult] = {
     for {
-      parent <- blockchain
+      parentHeader <- blockchain
         .getBlockHeaderByHash(block.header.parentHash)
         .toRight(MissingParentError) // Should not never occur because validated earlier
-      execResult <- executeBlockTransactions(block, parent)
+      initialWorld = InMemoryWorldStateProxy(
```

Comment:
I moved the creation on the `initialWorld` here because it the method `executeBlockTransactions` more testable.